### ISSUE TITLE
Do not complain about return type of dataProvider in test

### ISSUE THREAD

**Issue by ruudk** [phpstan/phpstan-phpunit#63]
Given I have the following test:
```php
    public function isActiveProvider() : iterable
    {
        yield 'title' => [true, new DateTime('2020-01-01 00:00:00'), null];
    }
```

PHPStan complains:
```
  Line   tests/ContractTest.php
 ------ -----------------------------------------------------------------------
  33     Method ContractTest::isActiveProvider() return type has no value type specified 
            in iterable type iterable.
         💡 Consider adding something like iterable<Foo> to the
         PHPDoc.
         You can turn off this check by setting
         checkMissingIterableValueType: false in your
         phpstan.neon.
```

There is no way to disable `checkMissingIterableValueType` per directory, so it would be nice if phpstan-phpunit would know this method was a dataprovider and just ignores it.

**Comment by ondrejmirtes** [comment]
The easiest way to do this is with `ignoreErrors`, similar to this:

```yaml
ignoreErrors:
    - '#Method .+::(data|provide).*\(\) return type has no value type specified in iterable type (array|iterable)\.#'
```
